src/provider.c

```diff
@@ -1158,7 +1158,7 @@ static const OSSL_ALGORITHM *
 p11prov_query_operation(void *provctx, int operation_id, int *no_cache)
 {
     P11PROV_CTX *ctx = (P11PROV_CTX *)provctx;
-    *no_cache = 0;
+    *no_cache = ctx->status == P11PROV_UNINITIALIZED ? 1 : 0;
```

Comment:
I know it looks redundant, but would you please explicitly assign this in each case as appropriate?
This way we make it very explicit which case we want to be cached or not 
(do still remove the initial initialization to 0).

Reply:
Done and force pushed.